MoltenVK/MoltenVK/Commands/MVKCmdDraw.mm

```diff
@@ -90,6 +97,34 @@
         // Thus, we can use a simple temporary buffer to stand in for the index buffer here.
         const auto idxSize = mvkMTLIndexTypeSizeInBytes((MTLIndexType)_binding.mtlIndexType);
         _binding.mtlBuffer = cmdEncoder->getTempMTLBuffer(idxSize)->_mtlBuffer;
+    } else if (_uint8) {
+        // Copy 8-bit indices into 16-bit index buffer compatible with Metal.
+        const uint32_t numIndices = (uint32_t)_binding.mtlBuffer.length - (uint32_t)_binding.offset;
+        auto* uint16Buf = cmdEncoder->getTempMTLBuffer(numIndices * 2);
+
+        if (_binding.mtlBuffer.storageMode != MTLStorageModePrivate) {
+            // Index buffer is CPU visible, so we can do a quick copy without breaking the render pass.
+            auto* uint16BufPtr = (uint16_t*)uint16Buf->getContents();
+            auto* uint8BufPtr = (uint8_t*)[_binding.mtlBuffer contents];
+            for (uint32_t i = 0; i < numIndices; i++) {
+                const auto idx = uint8BufPtr[_binding.offset + i];
+                uint16BufPtr[i] = idx == 0xFF ? 0xFFFF : idx;
+            }
+        } else {
+            // Index buffer is private, so we must do the conversion on the GPU.
+            cmdEncoder->encodeStoreActions(true);
+            id<MTLComputeCommandEncoder> mtlComputeEnc = cmdEncoder->getMTLComputeEncoder(kMVKCommandConvertUint8Indices, true);
+            [mtlComputeEnc setComputePipelineState: cmdEncoder->getCommandEncodingPool()->getConvertUint8IndicesMTLComputePipelineState()];
+            [mtlComputeEnc setBuffer:_binding.mtlBuffer offset: _binding.offset atIndex: 0];
+            [mtlComputeEnc setBuffer:uint16Buf->_mtlBuffer offset: 0 atIndex: 1];
+            [mtlComputeEnc setBytes: &numIndices length: sizeof(numIndices) atIndex: 2];
+            [mtlComputeEnc dispatchThreadgroups: MTLSizeMake(1, 1, 1) threadsPerThreadgroup: MTLSizeMake(1, 1, 1)];
```

Comment:
See my comments below about using `thread_position_in_grid` to let the GPU do what it does best.